samza-core/src/test/java/org/apache/samza/zk/TestZkJobCoordinator.java

```diff
@@ -62,12 +68,15 @@
 import static org.mockito.Mockito.verifyZeroInteractions;
 import static org.mockito.Mockito.when;
 
+@RunWith(PowerMockRunner.class)
```

Comment:
Why does this need powermock? Possible to refactor to not need this?

Reply:
This was required for mocking the static methods of `JobModelUtil` class in `ZkJobCoordinator` execution sequence.

Reply:
Question for @prateekm and @cameronlee314 (unrelated to this specific PR, but related to this comment)
In prior code reviews, @cameronlee314 mentioned that private methods should be static whenever possible so we can test/validate such methods easier. However, the only way to verify static methods are called in a test is to use PowerMock. What should be the proper convention for testing these type of static methods if PowerMock is discouraged?

Reply:
Sorry, I think I might have been a little unclear previously. By "validate", I was referring to when someone was checking correctness while reading the code, or having the compiler help to enforce certain usage. The "private static" methods aren't going to help with unit testing.
That said, it should be unnecessary to test "private static" methods directly since tests should theoretically only test against the public APIs of a class (although there are lots of exceptions to this, such as package-private VisibleForTesting methods). In this case, it is a static method being mocked, so we aren't directly testing a static method here. As @prateekm mentioned, verifications can be done against the objects that the static method works on. Or maybe if the static method is very complicated, it shouldn't be a static method, and it should be an object which can then be passed as a dependency to the object-under-test.

Reply:
Cool makes sense. Thanks for clarifying!

Reply:
> Prefer not using PowerMock and mocking statics

Done. Removed usages of `PowerMock` by mocking individual methods using `Mockito`.